File: elixir/sum-of-multiples/sum_of_multiples.exs
defmodule SumOfMultiples do
  def to(limit, multiples) do
    Enum.filter(1..(limit-1), fn n -> is_multiple_of?(n, multiples) end)
    |> Enum.sum
  end

  defp is_multiple_of?(number, multiples) when is_list(multiples) do
    Enum.any?(multiples, fn m -> is_multiple_of?(number, m) end)
  end

  defp is_multiple_of?(number, multiple), do: rem(number, multiple) == 0
end

File: erlang/sum-of-multiples/src/sum_of_multiples.erl
-module(sum_of_multiples).

-export([sumOfMultiples/2, test_version/0]).

sumOfMultiples(Multiples, N) ->
  Range = lists:seq(1, N - 1),
  Valid = lists:filter(fun (X) -> is_multiple(Multiples, X) end, Range),
  lists:sum(Valid).

is_multiple(Multiples, N) ->
  lists:any(fun (M) -> N rem M == 0 end, Multiples).

test_version() -> 1.
